src/game/Spell.cpp

```diff
@@ -2982,9 +2982,12 @@ void Spell::cast(bool skipCheck)
     // CAST SPELL
     SendSpellCooldown();
 
-    TakePower();
-    TakeReagents();                                         // we must remove reagents before HandleEffects to allow place crafted item in same slot
-    TakeAmmo();
+    if (!m_IsTriggeredSpell)
```

Comment:
Actually, I was talking about this: https://github.com/cmangos/mangos-tbc/commit/47ee89b33291d37b6dee9fab7f7a51f07821a3b1